src/transformers/models/perceiver/convert_perceiver_haiku_to_pytorch.py

```diff
@@ -0,0 +1,476 @@
+            state_dict["perceiver." + name] = torch.from_numpy(param)
+        else:
+            state_dict[name] = torch.from_numpy(param)
+
+
+@torch.no_grad()
+def convert_perceiver_checkpoint(pickle_file, pytorch_dump_folder_path, task="MLM"):
+    """
+    Copy/paste/tweak model's weights to our Perceiver structure.
+    """
+
+    # load parameters as FlatMapping data structure
+    with open(pickle_file, "rb") as f:
+        checkpoint = pickle.loads(f.read())
+
+    state = None
+    if isinstance(checkpoint, dict) and task in [
+        "image_classification",
+        "image_classification_fourier",
+        "image_classification_conv",
+    ]:
+        # the image classification_conv checkpoint also has batchnorm states (running_mean and running_var)
+        params = checkpoint["params"]
+        state = checkpoint["state"]
+    else:
+        params = checkpoint
+
+    # turn into initial state dict
+    state_dict = dict()
+    for scope_name, parameters in hk.data_structures.to_mutable_dict(params).items():
+        for param_name, param in parameters.items():
+            state_dict[scope_name + "/" + param_name] = param
+
+    if state is not None:
+        # add state variables
+        for scope_name, parameters in hk.data_structures.to_mutable_dict(state).items():
+            for param_name, param in parameters.items():
+                state_dict[scope_name + "/" + param_name] = param
+
+    # rename keys
+    rename_keys(state_dict, task=task)
+
+    # load HuggingFace model
+    config = PerceiverConfig()
+    subsampling = None
+    repo_id = "datasets/huggingface/label-files"
+    if task == "MLM":
+        config.qk_channels = 8 * 32
+        config.v_channels = 1280
+        model = PerceiverForMaskedLM(config)
+    elif "image_classification" in task:
+        config.num_latents = 512
+        config.d_latents = 1024
+        config.d_model = 512
+        config.num_blocks = 8
+        config.num_self_attends_per_block = 6
+        config.num_cross_attention_heads = 1
+        config.num_self_attention_heads = 8
+        config.qk_channels = None
+        config.v_channels = None
+        # set labels
+        config.num_labels = 1000
+        filename = "imagenet-1k-id2label.json"
+        id2label = json.load(open(cached_download(hf_hub_url(repo_id, filename)), "r"))
+        id2label = {int(k): v for k, v in id2label.items()}
+        config.id2label = id2label
+        config.label2id = {v: k for k, v in id2label.items()}
+        if task == "image_classification":
+            config.image_size = 224
+            model = PerceiverForImageClassification(config)
+        elif task == "image_classification_fourier":
+            config.d_model = 261
+            model = PerceiverForImageClassificationFourier(config)
+        elif task == "image_classification_conv":
+            config.d_model = 322
+            model = PerceiverForImageClassificationConvProcessing(config)
```

Comment:
I don't think *task* is the appropriate argument to distinguish between models. The task is image classification in all cases, the model should be picked with the architectures field of the config.

Reply:
This is just the conversion script, there's no task attribute in the config.

Reply:
Then can we rename this to something different of task?

Reply:
Ok, renamed to "architecture".